hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieCDCLogger.java

```diff
@@ -183,15 +183,16 @@ public void put(HoodieRecord hoodieRecord,
   private void flushIfNeeded(Boolean force) {
     if (force || numOfCDCRecordsInMemory.get() * averageCDCRecordSize >= maxBlockSize) {
       try {
-        List<HoodieRecord> records = cdcData.values().stream()
```

Comment:
The `values` has risk of OOM excepion, might need an improvement.

Reply:
Exactly, will have a deeper look.